src/overlays/actors/ovl_Door_Killer/z_door_killer.c

```diff
@@ -68,41 +84,438 @@ static ColliderJntSphInit D_80996000 = {
+            velocity.x = pos.x * 0.2f;
+            velocity.z = pos.z * 0.2f;
+            accel.x = -(velocity.x) * 0.1f;
+            accel.z = -(velocity.z) * 0.1f;
+            pos.x += this->actor.world.pos.x;
+            pos.y += this->actor.world.pos.y;
+            pos.z += this->actor.world.pos.z;
+            func_8002865C(globalCtx, &pos, &velocity, &accel, 300, 30);
+        }
+    }
+    if (!(this->hasHitPlayerOrGround & 1)) {
+        Vec3f playerPosRelToDoor;
+        Player* player = PLAYER;
+        func_8002DBD0(&this->actor, &playerPosRelToDoor, &player->actor.world.pos);
+        if ((fabsf(playerPosRelToDoor.y) < 20.0f) && (fabsf(playerPosRelToDoor.x) < 20.0f) &&
+            (playerPosRelToDoor.z < 100.0f) && (playerPosRelToDoor.z > 0.0f)) {
+            this->hasHitPlayerOrGround |= 1;
+            func_8002F6D4(globalCtx, &this->actor, 6.0f, this->actor.yawTowardsPlayer, 6.0f, 16);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_KDOOR_HIT);
+            func_8002F7DC(&player->actor, NA_SE_PL_BODY_HIT);
+        }
+    }
+    if (!(this->hasHitPlayerOrGround & 1) && (this->timer == 2)) {
+        this->hasHitPlayerOrGround |= 1;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_KDOOR_HIT_GND);
+    }
+}
+
+/**
+ * Wobble around, signifying the door is about to fall over. Does not set AC and so cannot be destroyed during this.
+ */
+void DoorKiller_Wobble(DoorKiller* this, GlobalContext* globalCtx) {
+    s16 rotation;
+    s32 i;
+
+    if ((this->timer == 16) || (this->timer == 8)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_KDOOR_WAVE);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995020.s")
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        this->actionFunc = DoorKiller_FallOver;
+        this->timer = 8;
+        this->hasHitPlayerOrGround &= ~1;
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_809951C4.s")
+    rotation = Math_SinS(this->timer * 0x2000) * this->timer * 100.0f;
+    for (i = 2; i < 9; i++) {
+        this->jointTable[i].y = rotation;
+    }
+    rotation = (u16)(s16)(-Math_CosS(this->timer * 0x1000) * 1000.0f) + 1000;
+    for (i = 2; i < 9; i++) {
+        this->jointTable[i].z = rotation;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_80995284.s")
+/**
+ * Idle while the player attempts to open the door and then begin to wobble
+ */
+void DoorKiller_WaitBeforeWobble(DoorKiller* this, GlobalContext* globalCtx) {
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        this->timer = 16;
+        this->actionFunc = DoorKiller_Wobble;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Killer/func_809952B8.s")
+void DoorKiller_Wait(DoorKiller* this, GlobalContext* globalCtx) {
+    Player* player;
```

Comment:
```suggestion
    Player* player = PLAYER;
```
should be fine to put this in the declaration.